### PR TITLE
Add custom collate functions for the `DataLoader`

### PR DESCRIPTION
**Pull request by emcastillo**
When using DataLoaders to feed wrapped chainer models in ignite, there is the need for the data to be converted to the current device array in chainer as all the computation graph must be done in chainer.

### CODE REVIEW

**Comment by emcastillo** [comment]
I think it is possible to have a single collate function that converts to the actual chainer device

File: chainer_pytorch_migration/ignite/collate.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import torch
+import cupy
+import numpy
+import re
+from torch._six import container_abcs, string_classes, int_classes
```
Comment (asi1024) [inline chainer_pytorch_migration/ignite/collate.py]: We don't have to consider using Python2 anymore.

File: chainer_pytorch_migration/ignite/collate.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import torch
+import cupy
```
Comment (msakai) [inline chainer_pytorch_migration/ignite/collate.py]: How about using the idiom?
```python
try:
    import cupy
    _cupy_import_error = None
except ImportError as e:
    _cupy_import_error = e
```

**Comment by emcastillo** [comment]
Redid it using cpm as array to share the tensor memory and use the torch default collate function

File: tests/test_collate.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import torch
+
+import chainer_pytorch_migration.ignite as cpm_ignite
+
+
+def test_collate():
+    collate = cpm_ignite.collate_to_array
+    dl = torch.utils.data.DataLoader([1], collate_fn=collate)
+    for x in dl:
+        assert isinstance(x[0], numpy.ndarray)
```
Comment (asi1024) [inline tests/test_collate.py]: How about using `numpy.testing.assert_array_equal(x, [numpy.array(1)])`?

File: tests/test_collate.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import numpy
+import torch
+
+import chainer_pytorch_migration.ignite as cpm_ignite
+
+
+def test_collate():
+    collate = cpm_ignite.collate_to_array
+    dl = torch.utils.data.DataLoader([1], collate_fn=collate)
```
Comment (asi1024) [inline tests/test_collate.py]: A 1-size array seems not suitable for the test case.

**Comment by emcastillo** [comment]
PTAL

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**